### PR TITLE
Refactor/onboarding

### PR DESCRIPTION
**Pull request by francisdelgad**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
I used code climate to determine what to change [code climate link](https://codeclimate.com/github/thepracticaldev/dev.to/app/javascript/src/Onboarding.jsx/source#issue-3901b378ed0645b142a97b36ce464dc5)

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/src/Onboarding.jsx [hunk 1]
```diff
@@ -235,49 +246,52 @@ class Onboarding extends Component {
   }
 
   handleNextHover() {
-    if (this.state.pageNumber === 2 && this.state.users.length === 0) {
+    const { pageNumber, users } = this.statel;
```
Comment (maestromac) [inline app/javascript/src/Onboarding.jsx]: Looks like a typo here.
```suggestion
    const { pageNumber, users } = this.state;
```

File: app/javascript/src/Onboarding.jsx [hunk 2]
```diff
@@ -173,16 +179,15 @@ class Onboarding extends Component {
...
-          this.setState({
-            allTags: this.state.allTags.map(currentTag => {
+          this.setState(prevState => ({
+            allTags: prevState.allTags.map(currentTag => {
               const newTag = currentTag;
-              if (currentTag.name === tag.name) {
-                newTag.following = json.outcome === 'followed';
-              }
+              newTag.following =
+                currentTag.name === tag.name ? json.outcome : 'followed';
```
Comment (maestromac) [inline app/javascript/src/Onboarding.jsx]: The functionality here changed. `newTag`'s `following` should be a be a boolean and not a string.

File: app/javascript/src/Onboarding.jsx [hunk 3]
```diff
@@ -191,42 +196,48 @@ class Onboarding extends Component {
...
-    const newCheckedUsers = this.state.checkedUsers.slice();
-    if (this.state.checkedUsers.indexOf(user) > -1) {
-      const index = newCheckedUsers.indexOf(user);
-      newCheckedUsers.splice(index, 1);
-    } else {
-      newCheckedUsers.push(user);
-    }
+    const { checkedUsers } = this.state;
+    const index = checkedUsers.indexOf(user);
+    const newCheckedUsers = (index > -1) ? [...checkedUsers].splice(index, 1) : checkedUsers.concat(user);
```
Comment (maestromac) [inline app/javascript/src/Onboarding.jsx]: newCheckedUsers would return incorrect info here. Splice's output is what was just spliced and not the whole array.

```
let arr = [1,2,3,4,5]
let spliced = arr.splice(1,0)
// spliced = [2]
// arr = [1,3,4,5]

**Comment by francisdelgad** [comment]
@maestromac thanks for the review addressed each comment 👍

File: app/javascript/src/Onboarding.jsx [hunk 4]
```diff
@@ -182,8 +182,7 @@ class Onboarding extends Component {
           this.setState(prevState => ({
             allTags: prevState.allTags.map(currentTag => {
               const newTag = currentTag;
-              newTag.following =
-                currentTag.name === tag.name ? json.outcome : 'followed';
+              newTag.following = json.outcome === 'followed';
```
Comment (maestromac) [inline app/javascript/src/Onboarding.jsx]: This part changed the functionality. You need to still check if  `currentTag.name === tag.name`. I'll change this for you.

**Pull request merged**